src/PerformanceSensitiveAnalyzers/UnitTests/CallSiteImplicitAllocationAnalyzerTests.cs

```diff
@@ -52,6 +53,44 @@ public void ParamsWithObjects(params object[] args)
                 VerifyCS.Diagnostic(CallSiteImplicitAllocationAnalyzer.ParamsParameterRule).WithLocation(16, 20));
         }
 
+
+        [Fact, WorkItem(3272, "https://github.com/dotnet/roslyn-analyzers/issues/3272")]
+        public async Task EmptyParamsWithNetFramework45()
+        {
+            await new VerifyCS.Test
+            {
+                ReferenceAssemblies = ReferenceAssemblies.NetFramework.Net45.Default,
+                TestState =
+                {
+                    Sources =
+                    {
+                        @"
+using System;
+using Roslyn.Utilities;
+
+public class MyClass
+{
+    [PerformanceSensitive(""uri"")]
+    public void Testing()
+    {
+        Params(); // allocation
+    }
+
+    public void Params(params int[] args)
+    {
+    }
+}",
+                        ("PerformanceSensitiveAttribute.cs", VerifyCS.PerformanceSensitiveAttributeSource)
+                    },
+                    ExpectedDiagnostics =
+                    {
+                        VerifyCS.Diagnostic(CallSiteImplicitAllocationAnalyzer.ParamsParameterRule).WithLocation(10, 9),
```

Comment:
Alright, so now, the test is failing because no issue is raised. In the `CheckParam` method when debugging the net45 case and the other case I don't seem to see any difference:

https://github.com/dotnet/roslyn-analyzers/blob/master/src/PerformanceSensitiveAnalyzers/CSharp/CallSiteImplicitAllocationAnalyzer.cs#L79-L83

Reply:
The only thing I can see (with my knowledge) is to access to the referenced assemblies through the compilation and to check the .net version using `mscorlib`. Then it that's earlier than net45 we do raise. Is it what you are thinking about?

Reply:
The allocation is only avoided if `System.Array.Empty<T>()` is defined.